src/equations/shallow_water_multilayer_2d.jl

```diff
@@ -0,0 +1,827 @@
+Create a boundary state by reflecting the normal velocity component and keep
+the tangential velocity component unchanged. The boundary water height is taken from
+the internal value.
+For details see Section 9.2.5 of the book:
+- Eleuterio F. Toro (2001)
+  Shock-Capturing Methods for Free-Surface Shallow Flows
+  1st edition
+  ISBN 0471987662
+"""
+@inline function Trixi.boundary_condition_slip_wall(u_inner,
+                                                    normal_direction::AbstractVector,
+                                                    x, t,
+                                                    surface_flux_function,
+                                                    equations::ShallowWaterMultiLayerEquations2D)
+    # normalize the outward pointing direction
+    normal = normal_direction / norm(normal_direction)
+
+    # Extract internal values
+    h = waterheight(u_inner, equations)
+    h_v1, h_v2 = momentum(u_inner, equations)
+    b = u_inner[end]
+
+    # u_boundary = SVector(h..., -hv..., b)
+
+    # compute the normal velocity
+    u_normal = normal[1] * h_v1 + normal[2] * h_v2
+
+    # create the "external" boundary solution state
+    u_boundary = SVector(h...,
+                         (h_v1 - 2.0 * u_normal * normal[1])...,
+                         (h_v2 - 2.0 * u_normal * normal[2])...,
+                         b)
+
+    # calculate the boundary flux
+    flux = surface_flux_function(u_inner, u_boundary, normal_direction, equations)
+
+    return flux
+end
+
+"""
+    boundary_condition_slip_wall(u_inner, orientation, direction, x, t,
+                                 surface_flux_function, equations::ShallowWaterMultiLayerEquations2D)
+
+Should be used together with [`TreeMesh`](@ref).
+"""
+@inline function Trixi.boundary_condition_slip_wall(u_inner, orientation,
+                                                    direction, x, t,
+                                                    surface_flux_function,
+                                                    equations::ShallowWaterMultiLayerEquations2D)
+    # Extract internal values
+    h = waterheight(u_inner, equations)
+    h_v1, h_v2 = momentum(u_inner, equations)
+    b = u_inner[end]
+
+    ## get the appropriate normal vector from the orientation
+    if orientation == 1
+        u_boundary = SVector(h..., -h_v1..., h_v2..., b)
+    else # orientation == 2
+        u_boundary = SVector(h..., h_v1..., -h_v2..., b)
+    end
+
+    # Calculate boundary flux
+    if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+        flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+    else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+        flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+    end
+
+    return flux
+end
+
+# Calculate 1D advective portion of the flux for a single point
+# Note, the bottom topography has no flux
```

Comment:
```suggestion
# Calculate 2D advective portion of the flux for a single point
# Note, the bottom topography has no flux
```

Reply:
Done in https://github.com/trixi-framework/TrixiShallowWater.jl/pull/40/commits/2eec19df7e184318796af5c0c14f2be8673918ab